src/meta/src/controller/streaming_job.rs

```diff
@@ -288,11 +288,18 @@ impl CatalogController {
             }
         }
 
+        // get dependent secret ref.
+        // XXX: A relation can ref a secret more than 1 time.
```

Comment:
Hmmm, perhaps we only need to record the dependency once.

Reply:
This is actually a tricky part. Unlike MV on MV, where we can't alter MV to make it depend on more/less MV, we should be able to alter MV to make it depend on more/less/another secret. Do you think in this case it's still better we record the dependency once? (I'm ok with both)

Reply:
> we should be able to alter MV to make it depend on more/less/another secret.
🤔  Make it depend on another secret, this requirement seems to have a bit of possibility. Anyway I just find it a bit odd to record two identical entries in the object dependency table.

Reply:
Oh, the records in system catalog `rw_catalog.rw_depend` are also from this system table. There will be some duplicate records if we insert more than once, which is quite confusing.

Reply:
Pushed a change. PTAL